### PR TITLE
Fix #2557: Add tests for worker_util.py for RL workers

### LINKED ISSUE #2557: Add tests for new RL Worker files
#### Scenario:
Recently, new files for RL workers have been added:
1. [rl_submission_worker.py](https://github.com/Cloud-CV/EvalAI/blob/master/scripts/workers/rl_submission_worker.py)
2. [worker_util.py](https://github.com/Cloud-CV/EvalAI/blob/master/scripts/workers/worker_util.py)

#### Solution
We will need new tests for these.

> warning note: the concerned part is still under development. Therefore, tests should be added for utility functions first (which are guaranteed to be used), and then those for the newer part.

### PR DESCRIPTION
**Pull request by nikochiko**
Add tests for worker_util.py in RL workers

### CODE REVIEW

File: tests/unit/rl_worker/test_worker_util.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                      body=requests.exceptions.RequestException(self.example_error_description))
+
+        with self.assertRaises(requests.exceptions.RequestException):
+            interface.make_request(url, "GET")
+
+        mock_logger.assert_called_with("The worker is not able to establish connection with EvalAI")
+
+    def test_get_message_from_sqs_queue(self):
+        url = "{}{}".format(evalai_api_server, self.get_message_from_sqs_queue_url(queue_name))
+        with mock.patch("__main__.EvalAI_Interface.make_request", return_value=self.success_response) as mock_make_request:
```
Comment (nikochiko) [inline tests/unit/rl_worker/test_worker_util.py]: @vkartik97 @RishabhJain2018 I am currently stuck on this issue since a few days.
The tests for `make_request` are working correctly, so I wanted to mock make_request in other tests, but this always throws an error. I tried using several different ways to patch like `patch("__main__.interface.make_request")` and simply `patch(interface.make_request)` which should work according to docs but the travis throws a `ModuleNotFoundError` with the message `__main__ is not a  package`.  
Is there a workaround for this or should we go with using the `responses` library to write tests?

**Comment by nikochiko** [comment]
@RishabhJain2018 @vkartik97 I have added the tests, PTAL.

File: scripts/workers/worker_util.py [hunk 2]
```diff
@@ -54,8 +54,8 @@ def delete_message_from_sqs_queue(self, receipt_handle):
         url = URLS.get("delete_message_from_sqs_queue").format(self.QUEUE_NAME)
         url = self.return_url_per_environment(url)
         data = {"receipt_handle": receipt_handle}
-        response = self.make_request(url, "POST", data)  # noqa
-        return response.status_code
+        response = self.make_request(url, "POST", data)
+        return response
```
Comment (vkartik97) [inline scripts/workers/worker_util.py]: Please remove this change, as it is being handled in https://github.com/Cloud-CV/EvalAI/pull/2575

File: scripts/workers/worker_util.py [hunk 3]
```diff
@@ -85,7 +85,7 @@ def get_challenge_phase_by_pk(self, challenge_pk, challenge_phase_pk):
         response = self.make_request(url, "GET")
         return response
 
-    def update_submission_data(self, data, challenge_pk, submission_pk):
+    def update_submission_data(self, data, challenge_pk):
```
Comment (vkartik97) [inline scripts/workers/worker_util.py]: Can you make a separate PR for this change as it is not part of `Add tests for worker_util.py`

File: scripts/workers/worker_util.py (hunk 2 above)
Comment (nikochiko) [reply scripts/workers/worker_util.py]: @vkartik97 :+1: I will also remove the test for `delete_message_for_sqs_queue`. It will cause an error without the correction so I think we can add it after the fix in #2575 is applied.

Comment (vkartik97) [reply scripts/workers/worker_util.py]: Sure 💯

File: scripts/workers/worker_util.py (hunk 3 above)
Comment (nikochiko) [reply scripts/workers/worker_util.py]: Great! @takitsuse had opened an PR already: https://github.com/Cloud-CV/EvalAI/pull/2611

Comment (nikochiko) [reply scripts/workers/worker_util.py]: Done here! :+1:

File: scripts/workers/worker_util.py (hunk 2 above)
Comment (nikochiko) [reply scripts/workers/worker_util.py]: @vkartik97 Done! 👍 Also verified that the Travis CI is passing.

**Comment by nikochiko** [comment]
@vkartik97 As the PR (https://github.com/Cloud-CV/EvalAI/pull/2575) with corrections in delete_message_from_sqs_queue has been merged, I have brought back the test for it :+1:.
cc: @RishabhJain2018

File: tests/unit/rl_worker/test_worker_util.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    def test_update_submission_data_partially(self):
+        url = "{}{}".format(EVALAI_API_SERVER, self.update_submission_status_url(self.challenge_pk))
+        data = {"test_field": "new_value"}
+        response = self.interface.update_submission_status(data, self.challenge_pk)
+
+        self.mock_make_request.assert_called_with(url, "PATCH", data=data)
+        self.assertEqual(response, self.success_response)
+
+    def test_delete_message_from_sqs_queue(self):
+        test_receipt_handle = "MbZj6wDWli+JvwwJaBV+3dcjk2YW2vA3+STFFljTM8tJJg6HRG6PYSasuWXPJB+Cw"
```
Comment (vkartik97) [inline tests/unit/rl_worker/test_worker_util.py]: Is this random string? If yes, how about using replacing hard coded string with a randomize function for alphanumeric and symbols?

Comment (nikochiko) [reply tests/unit/rl_worker/test_worker_util.py]: Nice idea! I will create a helper a helper for that.

Comment (nikochiko) [reply tests/unit/rl_worker/test_worker_util.py]: Done! 👍
